### ISSUE TITLE
Pb count_island

### ISSUE THREAD

**Issue by RoseChantelune** [yyang42/speedrun#1]
Le programme donne une erreur de compilation avec les 3 flags -W : variable ac non utilise dans le main.
Quand je lance le programme avec une map pas super grosse le programme segfault.
Je precise que je suis sur Linux Mint.
![capture-1](https://cloud.githubusercontent.com/assets/7979458/5975069/128b11b4-a87c-11e4-8681-1ab565560111.png)

**Comment by PierreLvx** [comment]
Version de GCC?

**Comment by RoseChantelune** [comment]
gcc (Ubuntu/Linaro 4.8.1-10ubuntu8) 4.8.1

 

Sarah COUDERT

 
 

Envoyé: vendredi 30 janvier 2015 à 12:33
De: "Pierre Lavaux" notifications@github.com
À: yyang42/speedrun speedrun@noreply.github.com
Cc: SarahCoudert sarahcoudert@gmx.fr
Objet: Re: [speedrun] Pb count_island (#1)

Version de GCC?

**Comment by PierreLvx** [comment]
Dac. Pour `ac`, le correctif est évident.

Tu peux joindre ta map de test?

**Comment by RoseChantelune** [comment]
La voici en dur : (je ne peux pas joindre de fichier text...)
........XXX................................................................................
........XXX................................................................................
.....XXXXX............................................XXXXXXX....XXXXX.....................
.................................................XXXXXXXX....XXXXX...XXXXX.................
....................................................XXXXXXXXXX..XXXXXX.....................
...........XXXXXXXX..................XXX...................................................
................XXX....................X...................................................
.......................................X...................................................
....................XXXXX..............X...................................................
.........................XXX...........X...................XXX.............................
.......................................X.................XXX.XXX...........................
.......................................X...............XXX.....XXX.........................
.......................................X.............XXX...XX....XXX.......................
................................XXXXXXXXXXXXXXXXXXXXXXX....XX..............................
...........................................................................................
...........................................................................................
...........................................................................................

**Comment by PierreLvx** [comment]
Merci. FYI, les dumps reposent sur GCC v4.9.0. J'ai testé avec ta map, et c'est OK.

Chez toi, segfault peu importe la map?

**Comment by RoseChantelune** [comment]
Oui, une map dont les lignes depassent 7 char segfault systematiquement, par contre aucune limitation en hauteur. Ca peut etre une pb lie a la gestion memoire de linux.
